### ISSUE TITLE
Amazon cloud credentials rarely showing up on Add Node Template modal

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#18225]
Version: Master 2.2 (2/19/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Add an ec2 node template. The first time you are prompted to create a cloud key. Make one.
- Create another ec2 node template. You have a small chance of seeing your created cloud key or a larger chance of not in the Add Node Template modal from the clusters page (drop-down). You might have to refresh the cluster page and try to re-add a cluster to see this.

**Result:**
Intermittently the amazon cloud key does not show up in the drop down
![image](https://user-images.githubusercontent.com/45179589/53032974-36425400-342d-11e9-894c-4415e445a896.png)

Cloud keys always show up on the Cloud Keys page though:
![image](https://user-images.githubusercontent.com/45179589/53033139-9df89f00-342d-11e9-82d3-e795d7887777.png)

**Other details that may be helpful:**
No errors in browser console.
When I inspect the node template I see it in the API. Such as https://(rancherserver)/v3/nodeTemplates?cloudCredentialId=cattle-global-data%3Aamazonkey shows the template.

**Comment by westlywright** [comment]
The only way I can repro this issue is by doing a refresh on the add cluster page, selecting amazon ec2 infra provider, and finally selecting the add template button.

**Comment by davidnuzik** [comment]
Version: Master 2.2 (2/19/19)
Confirmed fixed. I am no longer able to reproduce the issue. As Wes suggested I think we were refreshing the page or perhaps after a while of use close browser open again, and we could not see the cloudkey.
I tested these scenarios and I always see the cloudkey now. I smoke tested this area. Tried creating several keys, etc and make sure they all show up even after refresh / clear cache, etc.
